### PR TITLE
Feature/comments

### PR DESCRIPTION
**Pull request by johguentner**
[add comment entity, collection and endpoints](https://github.com/5am-code/laravel-notion-api/commit/6ddc29e9af9a550956b91c9ffdc499c51cee2b8f) 
- create comment entity with all available attributes
- create comment collection for a simple representation as list
- create endpoint for comment listing, based on ``block_id``
- create endpoint for comment creation, based on ``page_id`` or ``discussion_id``
- add comment endpoint access to ``Notion::class``

### CODE REVIEW

File: src/Endpoints/Comments.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    /**
+     * @param  string  $discussionId
+     * @return Comments
+     */
+    public function onDiscussion(string $discussionId): self
+    {
+        if ($this->pageId !== null) {
+            throw new HandlingException('You can only use ``->onDiscussion(...)`` or ``->onPage(...)``.');
```
Comment (mechelon) [inline src/Endpoints/Comments.php:66]: `onDiscussion()`

File: src/Endpoints/Comments.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace FiveamCode\LaravelNotionApi\Endpoints;
+
+use FiveamCode\LaravelNotionApi\Entities\Collections\CommentCollection;
+use FiveamCode\LaravelNotionApi\Entities\Comment;
```
Comment (mechelon) [inline src/Endpoints/Comments.php:6]: maybe use alias here?

File: src/Entities/Comment.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        parent::setResponseData($responseData);
+        if ($responseData['object'] !== 'comment') {
+            throw HandlingException::instance('invalid json-array: the given object is not a comment');
+        }
+        $this->fillFromRaw();
+    }
+
+    private function fillFromRaw(): void
+    {
+        parent::fillEntityBase();
```
Comment (mechelon) [inline src/Entities/Comment.php:80]: `fillBase()`?

File: src/Traits/HasTimestamps.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    protected ?User $createdBy = null;
+
+    /**
+     * @var User
+     */
+    protected ?User $lastEditedBy = null;
+
+    protected function fillTimestampableAttributes(): void
+    {
+        $this->fillCreatedTime();
```
Comment (mechelon) [inline src/Traits/HasTimestamps.php:42]: https://www.php.net/manual/en/function.class-uses.php

**Review by mechelon — APPROVED** [review]

File: src/Endpoints/Comments.php (hunk 1 above)
Comment (johguentner) [reply src/Endpoints/Comments.php]: ✅  done

File: src/Endpoints/Comments.php (hunk 2 above)
Comment (johguentner) [reply src/Endpoints/Comments.php]: ✅  done

File: src/Entities/Comment.php (hunk 3 above)
Comment (johguentner) [reply src/Entities/Comment.php]: ✅  done

Renamed to `fillEssentials()`

File: src/Traits/HasTimestamps.php (hunk 4 above)
Comment (johguentner) [reply src/Traits/HasTimestamps.php:42]: ✅  done

**Pull request merged**